.github/workflows/test.yml

```diff
@@ -109,3 +109,60 @@ jobs:
           bin_name: test_bin
           # Test setting just workdir.
           workdir: testdata
+  
+  action-test-smoke-test-pass:
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_TARBALL_NAME: action-test-smoke-test-pass.docker.tar
+    needs:
+      - action-test-prep
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        uses: ./
+        with:
+          version: 1.0.0
+          target: default
+          arch: amd64
+          tags: |
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-pass:tag1
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-pass:tag2
+          zip_artifact_name: test_bin.zip
+          bin_name: test_bin
+          dockerfile: testdata/Dockerfile
+          smoke_test: |
+            docker run "${IMAGE_NAME}" /bin/"${BIN_NAME}"
+
+  action-test-smoke-test-fail:
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_TARBALL_NAME: action-test-smoke-test-fail.docker.tar
+    needs:
+      - action-test-prep
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        id: docker-build
+        uses: ./
+        with:
+          version: 1.0.0
+          target: default
+          arch: amd64
+          tags: |
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-fail:tag1
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-fail:tag2
+          zip_artifact_name: test_bin.zip
+          bin_name: test_bin
+          dockerfile: testdata/Dockerfile
+          smoke_test: |
+            docker run "${IMAGE_NAME}"
```

Comment:
It's not clear that this command _should_ fail. I'm slightly surprised that it does in fact, I'll look into it as this should work really. Could you add an `exit 1` after this just to make absolutely sure that this script fails, and I'll separately work on getting the docker run to work!

Reply:
In fact, this failure case doesn't really need the `docker run` at all I think.. Just a pain `exit 1` is enough, because the passing smoke test test validates that the container runs already. Does that make sense?